src/compilersupport_p.h

```diff
@@ -33,6 +33,14 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#if __STDC_VERSION__ >= 201112L || __cplusplus >= 201103L || __cpp_static_assert >= 200410
+#  define cbor_static_assert(x)         static_assert(x, #x)
+#elif !defined(__cplusplus) && defined(__GNUC__) && (__GNUC__ * 100 + __GNUC_MINOR__ >= 406)
+#  define cbor_static_assert(x)         _Static_assert(x, #x)
+#else
+#  define cbor_static_assert(x)         ((void)sizeof(struct { int m : 2*!!(x) - 1; }))
```

Comment:
Nice. I usually see this defined in terms of `sizeof()` of an array with negative number of elements.